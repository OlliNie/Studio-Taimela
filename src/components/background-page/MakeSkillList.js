import React from 'react';
import MakeSkill from './MakeSkill';

export default function MakeSkillList({ skills }) {
  const skillList = skills.map((skill, i) => (
    <ul key={i}>
      <MakeSkill skill={skill} />
    </ul>
  ));
  return skillList;
}
