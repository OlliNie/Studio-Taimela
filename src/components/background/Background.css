.section{
  width:100%;
  min-height: 100vh;
  display:flex;
  flex-direction: column;
  text-align:center;
  background-image: url('../../assets/images/city.jpg');
  background-size: cover;
  background-repeat: no-repeat;
  background-attachment: fixed;

}
.ulEducation{
  padding-left: 0;
  display: flex;
  flex-direction:column;
  align-items: center;
  color:black;
 
  li:first-child{
    font-weight:700;
  
  }
  li{
    font-weight:100;
  }

}

.background{
  background-color:#F7F6F7;
  opacity: .9;
  margin: 0 1rem;
  padding:1rem;
  color:black;
}
.skills{
  display:flex;
  flex-direction:column;
  @media (min-width: 720px){
    flex-direction: row;
  }
  justify-content: center;
  text-align: left;
  ul{
    padding:0;
    margin:0;
  }

  
  h2{
    margin:0;
    font-size:1rem;
  }
}
