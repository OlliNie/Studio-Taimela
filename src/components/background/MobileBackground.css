.background{
  h2, h3, h4, h5, p{
    font-size:1rem;
  }
  text-align:center;
  h4{
    display:inline-block;
    margin: 0 .3rem 0 0;
  }
  h3, h5{
    margin:0;
  }
  h5{
    margin-left:.3rem;
  }
  
  p{
    margin:0;
  }
  ul{
    display:flex;
    margin-bottom:.2rem;
  }
  p{
    white-space: nowrap;
  }
  ul{
    justify-content: center;
    display:flex;
    flex-wrap: wrap;
  }
  @media(min-width: 1100px){
    display:flex;
    section:first-child{
      margin-right: 2rem;
      margin-left:auto;
    }
    section:last-child{
      margin-right:auto;
    }
  }
}
.section{
  margin-bottom:1rem;
  @media(min-width: 750px){
    text-align:left;
    ul{
      justify-content: left;
    }
  }

}
.check{
  color:#FFD67A
}