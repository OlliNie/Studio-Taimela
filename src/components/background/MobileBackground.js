import React from 'react';
import styles from './MobileBackground.css';

export default function MobileBackground(){
  return (
    <section className={styles.background}>
      <section className={styles.section}>
        <h3>Languages</h3>
        <ul>
          <li className={styles.check}>✓</li>
          <li><h4>English</h4></li>
          <li><p>Native, born in</p></li>
          <li><p>the USA</p></li>
        </ul>
        <ul>
          <li className={styles.check}>✓</li>
          <li><h4>Finnish</h4></li>
          <li><p>Mother tongue</p></li>
        </ul>
        <ul>
          <li className={styles.check}>✓</li>
          <li><h4>French</h4></li>
          <li><p>Conversational (B2)</p></li>

        </ul>
        <ul>
          <li className={styles.check}>✓</li>
          <li><h4>German</h4></li>
          <li><p>Conversational (B1)</p></li>
        </ul>
      </section>
      <section className={styles.section}>
        <h3>Certificates</h3>
        <ul>
          <li className={styles.check}>✓</li>
          <li><h4>TOGAF</h4></li>
          <li><p>Foundation</p></li>
        </ul>
        <ul>
          <li className={styles.check}>✓</li>
          <li><h4>ITIL</h4></li>
          <li><p>Certified</p></li>
        </ul>
        <ul>
          <li className={styles.check}>✓</li>
          <li><h4>Prince 2</h4></li>
          <li><p>Foundation</p></li>
        </ul>
        <ul>
          <li className={styles.check}>✓</li>
          <li><p>Trained in</p></li>
          <li><h5>Balanced</h5></li>
          <li><h5>Scorebard</h5></li>
        </ul>
      </section>
    </section>
  );
}
