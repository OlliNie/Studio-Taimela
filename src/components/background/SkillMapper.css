.symbol{
  color:gold;
}
.skill{
  display:flex;
  p{
    margin:0;
    padding-right:.4rem;
  }
}