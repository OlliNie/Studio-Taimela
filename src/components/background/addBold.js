import React from 'react';

export function addBold(string, index) {
  const array = string.split(' ');
  let withBoldArray = [];

  withBoldArray = array.map((word, wordIndex) => {
    if (index.includes(wordIndex)) {
      return (
        <li key={wordIndex}>
          <h4>{word}</h4>
        </li>
      );
    }
    return (
      <li key={wordIndex}>
        <p>{word}</p>
      </li>
    );
  });
  return withBoldArray;
}
