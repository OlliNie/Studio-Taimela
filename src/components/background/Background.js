import React from 'react';
import styles from './Background.css';
import { education, languages, certificates } from '../../text/background';
import Title from '../title/Title';
import SkillMapper from './SkillMapper';
import { FaLinkedin } from 'react-icons/fa';

export default function Background() {

  return (
    <section id='Background' className={[styles.backgroundPage, 'backgroundPage'].join(' ')}>
    
      <section className={styles.background}>
        <Title title='background' />

        <section className={styles.backgroundText}>
          <section>
            <ul className={styles.ulEducation}>
              <li>{education.degree}</li>
              <li>{education.school}</li>
            </ul>
          </section>
          <section className={styles.skills}>
            <SkillMapper title={languages.title} skills={languages.skills} />
            <SkillMapper title={certificates.title} skills={certificates.skills} />
          </section>
        </section>

      </section>
      <section className={styles.social}>
        <p>
            Find my detailed work experience on LinkedIn
        </p>
        <a alt="Linked In" aria-label= "Linked In" rel= "noopener noreferrer" target = '_blank' href='https://www.linkedin.com/in/virpinieminen/'>
          <FaLinkedin size={42} />
        </a>
      </section>



    </section>
  );
}
