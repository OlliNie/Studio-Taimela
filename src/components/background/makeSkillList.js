import React from 'react';
import { addBold } from './addBold';
import styles from './MobileBackground.css';

export function makeSkillList(skills) {
  const ul = skills.map((skill, i) => (
    <ul key={i}>
      <li className={styles.check}>âœ“</li>
      {addBold(skill.text, skill.bold)}
    </ul>
  ));
  return ul;
}
