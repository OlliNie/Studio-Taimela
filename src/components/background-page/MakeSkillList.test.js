import React from 'react';
import { shallow } from 'enzyme';
import MakeSkillList from './MakeSkillList';
import { languageSkills } from '../../text/background';

describe('MakeSkillList comonent', () => {
  it('renders a skill list', () => {
    const wrapper = shallow(<MakeSkillList skills={languageSkills} />);
    expect(wrapper).toMatchSnapshot();
  });
});
