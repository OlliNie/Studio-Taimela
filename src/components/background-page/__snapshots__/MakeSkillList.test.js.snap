// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MakeSkillList comonent renders a skill list 1`] = `
Array [
  <ul
    key="0"
  >
    <MakeSkill
      skill={
        Object {
          "bold": Array [
            0,
          ],
          "text": "English Native, born in the USA",
        }
      }
    />
  </ul>,
  <ul
    key="1"
  >
    <MakeSkill
      skill={
        Object {
          "bold": Array [
            0,
          ],
          "text": "Finnish Mother tongue",
        }
      }
    />
  </ul>,
  <ul
    key="2"
  >
    <MakeSkill
      skill={
        Object {
          "bold": Array [
            0,
          ],
          "text": "French Conversational (B2)",
        }
      }
    />
  </ul>,
  <ul
    key="3"
  >
    <MakeSkill
      skill={
        Object {
          "bold": Array [
            0,
          ],
          "text": "German Conversational (B1)",
        }
      }
    />
  </ul>,
]
`;
